FinanceManager.Account/Domain/Account.cs

```diff
@@ -0,0 +1,93 @@
+﻿using FinanceManager.Events.Models;
+using FinanceManager.TransportLibrary.Services;
+
+namespace FinanceManager.Account.Domain;
+
+public class Account
+{
+    public Guid Id { get; set; }
+    public string RequestId { get; set; } = null!;
+    public long UserId { get; set; }
+    public string Name { get; set; } = null!;
+    public decimal Balance { get; set; }
+    public Guid CurrencyId { get; set; }
+    public string? Icon { get; set; }
+    public string? Description { get; set; }
+
+    public bool ValidateAndUpdateBalance(
+        AccountLimit[] limits,
+        decimal changeValue,
+        DateTime changeDate,
+        Guid changeCategoryId,
+        Guid accountId,
+        string? userAddress,
+        string transactionId,
+        ILogger logger,
+        IMessagePublisher<NotificationSendEvent> notificationSendPublisher,
+        IMessagePublisher<ChangeStatisticsEvent> changeStatisticsPublisher)
+    {
+        var newBalanceValue = Balance + changeValue;
+        if (newBalanceValue < 0)
+        {
+            logger.LogWarning(
+                "Account with id {AccountId} balance less zero",
+                accountId);
+            if (userAddress != null)
+            {
+                notificationSendPublisher.Send(new NotificationSendEvent
```

Comment:
Please consider separate pub/sub queues inside application to offload this work as this piece is used in async context upstream; In return you can significantly decrease latency and potential locking as well as issues related to a thread starvation (sync-over-async issue).